### ISSUE TITLE
arrow unsupported chemistries (compatibility?) issue

### ISSUE THREAD

**Issue by maximilianpress** [PacificBiosciences/pbbioconda#81]
**Operating system**
I am using Amazon Linux 
(`Amazon Linux AMI release 2017.09
Kernel \r on an \m`)

**Package name**
I am running arrow (variantCaller) v2.3.2. 

**Conda environment**
```
# packages in environment at /home/ec2-user/miniconda3:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py27_0  
avro-python2              1.8.2                      py_1    bioconda
backports                 1.0                      py27_1  
backports.functools_lru_cache 1.5                      py27_1  
bcftools                  1.9                  h4da6232_0    bioconda
biopython                 1.72                     pypi_0    pypi
blas                      1.0                         mkl  
blasr                     5.3.2                hac9d22c_4    bioconda
blasr_libcpp              5.3.1                hac9d22c_3    bioconda
bzip2                     1.0.6                h14c3975_5  
ca-certificates           2018.12.5                     0  
certifi                   2018.11.29               py27_0  
cffi                      1.11.5           py27he75722e_1  
chardet                   3.0.4                    py27_1  
conda                     4.6.1                    py27_0  
conda-env                 2.6.0                         1  
cryptography              2.4.2            py27h1ba5d50_0  
curl                      7.63.0            hbc83047_1000  
cython                    0.28.5           py27hf484d3e_0  
enum34                    1.1.6                    py27_1  
futures                   3.2.0                    py27_0  
genomicconsensus          2.3.2                    py27_3    bioconda
h5py                      2.8.0            py27h989c5e5_3  
hdf5                      1.10.2               hba1933b_1  
htslib                    1.9                  hc238db4_4    bioconda
idna                      2.7                      py27_0  
intel-openmp              2019.0                      118  
ipaddress                 1.0.22                   py27_0  
iso8601                   0.1.12                   py27_1  
krb5                      1.16.1               h173b8e3_7  
libcurl                   7.63.0            h20c2e04_1000  
libdeflate                1.0                  h470a237_0    bioconda
libedit                   3.1.20170329         h6b74fdf_2  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 8.2.0                hdf63c60_1  
libgfortran-ng            7.3.0                hdf63c60_0  
libopenblas               0.3.3                h5a2b251_3  
libssh2                   1.8.0                h1ba5d50_4  
libstdcxx-ng              8.2.0                hdf63c60_1  
linecache2                1.0.0                    py27_0  
mkl                       2019.1                      144  
mkl_fft                   1.0.10                   py27_0    conda-forge
mkl_random                1.0.2                    py27_0    conda-forge
ncurses                   6.1                  hf484d3e_0  
numpy                     1.15.4           py27h7e9f1db_0  
numpy-base                1.15.4           py27hde5b4d6_0  
openblas                  0.3.3             h9ac9557_1001    conda-forge
openssl                   1.1.1a               h7b6447c_0  
pbalign                   0.3.2                    py27_0    bioconda
pbbam                     0.19.0               h6678c95_1    bioconda
pbcommand                 1.1.1                    py27_2    bioconda
pbcore                    1.6.5                    py27_0    bioconda
pip                       10.0.1                   py27_0  
pycosat                   0.6.3            py27h14c3975_0  
pycparser                 2.18                     py27_1  
pyopenssl                 18.0.0                   py27_0  
pysam                     0.15.1           py27h0380709_0    bioconda
pysocks                   1.6.8                    py27_0  
python                    2.7.15               h9bab390_6  
python-consensuscore      1.1.1            py27h02d93b8_2    bioconda
python-consensuscore2     3.1.0                    py27_2    bioconda
python-dateutil           2.7.5                    py27_0  
pytz                      2018.5                   py27_0  
readline                  7.0                  h7b6447c_5  
requests                  2.21.0                   py27_0  
ruamel_yaml               0.15.46          py27h14c3975_0  
samtools                  1.9                  h8ee4bcc_1    bioconda
setuptools                40.2.0                   py27_0  
six                       1.11.0                   py27_1  
sqlite                    3.26.0               h7b6447c_0  
tk                        8.6.8                hbc83047_0  
traceback2                1.4.0                    py27_0  
unittest2                 1.1.0                    py27_0  
urllib3                   1.24.1                   py27_0  
wheel                     0.31.1                   py27_0  
xz                        5.2.4                h14c3975_4  
yaml                      0.1.7                had09818_2  
zlib                      1.2.11               ha838bed_2  
```
**Describe the bug**
When I run arrow on bam files from pbalign with chemistry info (AFAICT) in the bam file, it is not detected by arrow apparently. 

Here is what I see when I `samtools view -h | grep "@PG"` the subreads and the pbalign bam from aligning the subreads to an assembly, showing what I understand to be information sufficient to infer the chemistry:
```
# subreads
@RG     ID:8691335f     PL:PACBIO       DS:READTYPE=SUBREAD;Ipd:CodecV1=ip;PulseWidth:CodecV1=pw;BINDINGKIT=101-365-900;SEQUENCINGKIT=101-309-400;BASECALLERVERSION=5.0.0;FRAMERATEHZ=80.000000 PU:m54215_180723_183421 PM:SEQUEL
# pbalign alignments
@RG     ID:442094c0     PL:PACBIO       DS:READTYPE=SUBREAD;Ipd:CodecV1=ip;PulseWidth:CodecV1=pw;BINDINGKIT=101-365-900;SEQUENCINGKIT=101-309-400;BASECALLERVERSION=5.0.0;FRAMERATEHZ=80.000000 PU:m54215_180725_114246 PM:SEQUEL
```

If I supply a (I believe wrong) chemistry via `-p S/P3-C3/5.0` when running arrow, I get a different error:
`ValueError: no field of name BaseFeatures`
Which is probably not surprising given that I have no idea what the expected inputs are for this. I added some print statements to arrow.py to try to reverse engineer this but was no wiser except that the above is apparently an expected string, it's just insufficient but it's not clear what else is required.

**Error message**
```
Arrow: unsupported chemistries found: (unknown)
Arrow: unsupported chemistries found: (unknown)
Traceback (most recent call last):
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/pbcommand/cli/core.py", line 138, in _pacbio_main_runner
    return_code = exe_main_func(*args, **kwargs)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 340, in args_runner
    return tr.main()
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 270, in main
    self._configureAlgorithm(options, peekFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 164, in _configureAlgorithm
    self._algorithmConfiguration = self._algorithm.configure(options, alnFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/arrow/arrow.py", line 236, in configure
    die("Arrow: unsupported chemistries found: ({0})".format(", ".join(sorted(unsupp))))
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/utils.py", line 16, in die
    raise DieException(msg)
DieException: Arrow: unsupported chemistries found: (unknown)
[ERROR] Arrow: unsupported chemistries found: (unknown)
Traceback (most recent call last):
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/pbcommand/cli/core.py", line 138, in _pacbio_main_runner
    return_code = exe_main_func(*args, **kwargs)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 340, in args_runner
    return tr.main()
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 270, in main
    self._configureAlgorithm(options, peekFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 164, in _configureAlgorithm
    self._algorithmConfiguration = self._algorithm.configure(options, alnFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/arrow/arrow.py", line 236, in configure
    die("Arrow: unsupported chemistries found: ({0})".format(", ".join(sorted(unsupp))))
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/utils.py", line 16, in die
    raise DieException(msg)
DieException: Arrow: unsupported chemistries found: (unknown)
```

There is additionally an error about numpy (not fixed by reinstalling arrow and/or numpy, switching python versions or py2/3 conda envs), but I have seen this error in the past and been able to get arrow to run successfully so I doubt it is directly relevant though I could be wrong:
```
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
```

Any help would be appreciated.

**To Reproduce**
I unfortunately cannot share the relevant data. I don't know how to reconstruct the error due to the convoluted dependencies of arrow and nature of PB sequencing data, and I don't know if what I see is only an epiphenomenon of some hidden issue. Based on [other issues I've found](https://github.com/PacificBiosciences/pbbioconda/issues/6) it looks like unsupported-chemistries might be more of a general complaining mode of arrow and less of an informative error about chemistry information.

**Expected behavior**
I expected arrow to run and be able to detect chemistry information from raw PB subreads. I am not sure whether the error is due to a failure to do this or a buried issue.

**Comment by pb-dseifert** [comment]
would you be able to share the raw and aligned data so we can reproduce this?

**Comment by maximilianpress** [comment]
[Here](https://drive.google.com/open?id=17x5EVyWfMJg9W92lBCgRJZ_7ujnRQRJN) is a subsample of raw+aligned data that reproduces the issue in my environment (as detailed above).

I can't share the full dataset or the reference, unfortunately.

Thanks, max

**Comment by pb-dseifert** [comment]
Your channels seems incorrectly setup, have you checked them against https://bioconda.github.io/#set-up-channels and run `conda update --all`?

**Comment by maximilianpress** [comment]
I tried various versions of what you suggest previously and it didn't seem to change anything. 

However I re-set up the conda channels as suggested and ran `conda update --all`. My `.condarc` looks like this:
```
channels:
  - conda-forge
  - bioconda
  - defaults
```

It produces the expected outputs:

```
-bash-4.2$ conda config --add channels defaults
Warning: 'defaults' already in 'channels' list, moving to the top
-bash-4.2$ conda config --add channels bioconda
Warning: 'bioconda' already in 'channels' list, moving to the top
-bash-4.2$ conda config --add channels conda-forge
Warning: 'conda-forge' already in 'channels' list, moving to the top
-bash-4.2$ conda update --all
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /home/ec2-user/miniconda3


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    asn1crypto-0.24.0          |        py27_1003         154 KB  conda-forge
    backports-1.0              |             py_2           4 KB  conda-forge
    backports.functools_lru_cache-1.5|             py_1           6 KB  conda-forge
    bcftools-1.9               |       h47928c2_1         813 KB  bioconda
    blasr-5.3.2                |       h82bacf8_5         291 KB  bioconda
    blasr_libcpp-5.3.1         |       h82bacf8_4         2.0 MB  bioconda
    bzip2-1.0.6                |    h14c3975_1002         415 KB  conda-forge
    cffi-1.12.0                |   py27h9745a5d_0         213 KB  conda-forge
    chardet-3.0.4              |        py27_1003         180 KB  conda-forge
    conda-4.6.3                |           py27_0         867 KB  conda-forge
    conda-env-2.6.0            |                1           2 KB  conda-forge
    cryptography-2.5           |   py27hb7f436b_0         639 KB  conda-forge
    curl-7.64.0                |       h646f8bb_0         143 KB  conda-forge
    cython-0.29.5              |   py27hf484d3e_0         2.1 MB  conda-forge
    enum34-1.1.6               |        py27_1001          56 KB  conda-forge
    futures-3.2.0              |        py27_1000          24 KB  conda-forge
    h5py-2.8.0                 |py27hfb8609e_1004         1.0 MB  conda-forge
    hdf5-1.10.3                |    hba1933b_1001         5.3 MB  conda-forge
    htslib-1.9                 |       h47928c2_5         1.5 MB  bioconda
    idna-2.8                   |        py27_1000         133 KB  conda-forge
    ipaddress-1.0.22           |             py_1          18 KB  conda-forge
    iso8601-0.1.12             |             py_1           9 KB  conda-forge
    krb5-1.16.3                |    hc83ff2d_1000         1.4 MB  conda-forge
    libcurl-7.64.0             |       h01ee5af_0         586 KB  conda-forge
    libdeflate-1.0             |       h14c3975_1          43 KB  bioconda
    libedit-3.1.20170329       |    hf8c457e_1001         172 KB  conda-forge
    linecache2-1.0.0           |             py_1          13 KB  conda-forge
    numpy-1.16.1               |py27_blas_openblash1522bff_0         4.2 MB  conda-forge
    pip-19.0.2                 |           py27_0         1.8 MB  conda-forge
    pycosat-0.6.3              |py27h14c3975_1001         103 KB  conda-forge
    pycparser-2.19             |             py_0          87 KB  conda-forge
    pyopenssl-19.0.0           |           py27_0          79 KB  conda-forge
    pysam-0.15.2               |   py27h1671916_1         2.3 MB  bioconda
    pysocks-1.6.8              |        py27_1002          22 KB  conda-forge
    python-2.7.15              |    h938d71a_1006        11.9 MB  conda-forge
    python-dateutil-2.8.0      |             py_0         219 KB  conda-forge
    pytz-2018.9                |             py_0         229 KB  conda-forge
    requests-2.21.0            |        py27_1000          83 KB  conda-forge
    ruamel_yaml-0.15.71        |py27h14c3975_1000         249 KB  conda-forge
    samtools-1.9               |       h57cc563_6         636 KB  bioconda
    setuptools-40.8.0          |           py27_0         630 KB  conda-forge
    six-1.12.0                 |        py27_1000          21 KB  conda-forge
    traceback2-1.4.0           |           py27_0          28 KB  conda-forge
    unittest2-1.1.0            |             py_0          68 KB  conda-forge
    urllib3-1.24.1             |        py27_1000         146 KB  conda-forge
    wheel-0.33.0               |           py27_0          34 KB  conda-forge
    yaml-0.1.7                 |    h14c3975_1001          78 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        40.9 MB

The following packages will be UPDATED:

  asn1crypto            pkgs/main::asn1crypto-0.24.0-py27_0 --> conda-forge::asn1crypto-0.24.0-py27_1003
  backports          pkgs/main/linux-64::backports-1.0-py2~ --> conda-forge/noarch::backports-1.0-py_2
  bcftools                                   1.9-h4da6232_0 --> 1.9-h47928c2_1
  blas                              pkgs/main::blas-1.0-mkl --> conda-forge::blas-1.1-openblas
  blasr                                    5.3.2-hac9d22c_4 --> 5.3.2-h82bacf8_5
  blasr_libcpp                             5.3.1-hac9d22c_3 --> 5.3.1-h82bacf8_4
  bzip2                   pkgs/main::bzip2-1.0.6-h14c3975_5 --> conda-forge::bzip2-1.0.6-h14c3975_1002
  cffi                pkgs/main::cffi-1.11.5-py27he75722e_1 --> conda-forge::cffi-1.12.0-py27h9745a5d_0
  chardet                   pkgs/main::chardet-3.0.4-py27_1 --> conda-forge::chardet-3.0.4-py27_1003
  conda                                        4.6.1-py27_0 --> 4.6.3-py27_0
  cryptography                      2.3.1-py27hb7f436b_1000 --> 2.5-py27hb7f436b_0
  curl                                 7.63.0-h646f8bb_1000 --> 7.64.0-h646f8bb_0
  cython             pkgs/main::cython-0.28.5-py27hf484d3e~ --> conda-forge::cython-0.29.5-py27hf484d3e_0
  enum34                     pkgs/main::enum34-1.1.6-py27_1 --> conda-forge::enum34-1.1.6-py27_1001
  futures                   pkgs/main::futures-3.2.0-py27_0 --> conda-forge::futures-3.2.0-py27_1000
  h5py                 pkgs/main::h5py-2.8.0-py27h989c5e5_3 --> conda-forge::h5py-2.8.0-py27hfb8609e_1004
  hdf5                    pkgs/main::hdf5-1.10.2-hba1933b_1 --> conda-forge::hdf5-1.10.3-hba1933b_1001
  htslib                                     1.9-hc238db4_4 --> 1.9-h47928c2_5
  idna                           pkgs/main::idna-2.7-py27_0 --> conda-forge::idna-2.8-py27_1000
  intel-openmp                                   2019.0-118 --> 2019.1-144
  ipaddress          pkgs/main/linux-64::ipaddress-1.0.22-~ --> conda-forge/noarch::ipaddress-1.0.22-py_1
  krb5                                 1.16.2-hc83ff2d_1000 --> 1.16.3-hc83ff2d_1000
  libcurl                              7.63.0-h01ee5af_1000 --> 7.64.0-h01ee5af_0
  libdeflate                                 1.0-h470a237_0 --> 1.0-h14c3975_1
  libedit            pkgs/main::libedit-3.1.20170329-h6b74~ --> conda-forge::libedit-3.1.20170329-hf8c457e_1001
  libffi                 pkgs/main::libffi-3.2.1-hd88cf55_4 --> conda-forge::libffi-3.2.1-hf484d3e_1005
  linecache2         pkgs/main/linux-64::linecache2-1.0.0-~ --> conda-forge/noarch::linecache2-1.0.0-py_1
  ncurses                 pkgs/main::ncurses-6.1-hf484d3e_0 --> conda-forge::ncurses-6.1-hf484d3e_1002
  numpy              pkgs/main::numpy-1.15.4-py27h7e9f1db_0 --> conda-forge::numpy-1.16.1-py27_blas_openblash1522bff_0
  pip                          pkgs/main::pip-10.0.1-py27_0 --> conda-forge::pip-19.0.2-py27_0
  pycosat            pkgs/main::pycosat-0.6.3-py27h14c3975~ --> conda-forge::pycosat-0.6.3-py27h14c3975_1001
  pycparser          pkgs/main/linux-64::pycparser-2.18-py~ --> conda-forge/noarch::pycparser-2.19-py_0
  pyopenssl              pkgs/main::pyopenssl-18.0.0-py27_0 --> conda-forge::pyopenssl-19.0.0-py27_0
  pysam                               0.15.1-py27h0380709_0 --> 0.15.2-py27h1671916_1
  pysocks                   pkgs/main::pysocks-1.6.8-py27_0 --> conda-forge::pysocks-1.6.8-py27_1002
  python                               2.7.15-h938d71a_1004 --> 2.7.15-h938d71a_1006
  python-dateutil    pkgs/main/linux-64::python-dateutil-2~ --> conda-forge/noarch::python-dateutil-2.8.0-py_0
  pytz               pkgs/main/linux-64::pytz-2018.5-py27_0 --> conda-forge/noarch::pytz-2018.9-py_0
  readline               pkgs/main::readline-7.0-h7b6447c_5 --> conda-forge::readline-7.0-hf8c457e_1001
  requests                pkgs/main::requests-2.21.0-py27_0 --> conda-forge::requests-2.21.0-py27_1000
  ruamel_yaml        pkgs/main::ruamel_yaml-0.15.46-py27h1~ --> conda-forge::ruamel_yaml-0.15.71-py27h14c3975_1000
  samtools                                   1.9-h8ee4bcc_1 --> 1.9-h57cc563_6
  setuptools            pkgs/main::setuptools-40.2.0-py27_0 --> conda-forge::setuptools-40.8.0-py27_0
  six                          pkgs/main::six-1.11.0-py27_1 --> conda-forge::six-1.12.0-py27_1000
  sqlite                pkgs/main::sqlite-3.26.0-h7b6447c_0 --> conda-forge::sqlite-3.26.0-h67949de_1000
  tk                         pkgs/main::tk-8.6.8-hbc83047_0 --> conda-forge::tk-8.6.9-h84994c4_1000
  urllib3                  pkgs/main::urllib3-1.24.1-py27_0 --> conda-forge::urllib3-1.24.1-py27_1000
  wheel                      pkgs/main::wheel-0.31.1-py27_0 --> conda-forge::wheel-0.33.0-py27_0
  xz                         pkgs/main::xz-5.2.4-h14c3975_4 --> conda-forge::xz-5.2.4-h14c3975_1001
  yaml                     pkgs/main::yaml-0.1.7-had09818_2 --> conda-forge::yaml-0.1.7-h14c3975_1001
  zlib                    pkgs/main::zlib-1.2.11-ha838bed_2 --> conda-forge::zlib-1.2.11-h14c3975_1004

The following packages will be SUPERSEDED by a higher-priority channel:

  backports.functoo~ pkgs/main/linux-64::backports.functoo~ --> conda-forge/noarch::backports.functools_lru_cache-1.5-py_1
  conda-env                              pkgs/main/linux-64 --> conda-forge/noarch
  iso8601            pkgs/main/linux-64::iso8601-0.1.12-py~ --> conda-forge/noarch::iso8601-0.1.12-py_1
  libgcc-ng           pkgs/main::libgcc-ng-8.2.0-hdf63c60_1 --> conda-forge::libgcc-ng-7.3.0-hdf63c60_0
  libgfortran-ng     pkgs/main::libgfortran-ng-7.3.0-hdf63~ --> conda-forge::libgfortran-ng-7.2.0-hdf63c60_3
  libstdcxx-ng       pkgs/main::libstdcxx-ng-8.2.0-hdf63c6~ --> conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
  traceback2                                      pkgs/main --> conda-forge
  unittest2          pkgs/main/linux-64::unittest2-1.1.0-p~ --> conda-forge/noarch::unittest2-1.1.0-py_0

The following packages will be DOWNGRADED:

  numpy-base                          1.15.4-py27hde5b4d6_0 --> 1.15.4-py27h2f8d375_0


Proceed ([y]/n)? y


Downloading and Extracting Packages
conda-4.6.3          | 867 KB    | ################################################################################################################################################################################################################################### | 100% 
libdeflate-1.0       | 43 KB     | ################################################################################################################################################################################################################################### | 100% 
urllib3-1.24.1       | 146 KB    | ################################################################################################################################################################################################################################### | 100% 
cffi-1.12.0          | 213 KB    | ################################################################################################################################################################################################################################### | 100% 
blasr-5.3.2          | 291 KB    | ################################################################################################################################################################################################################################### | 100% 
pysam-0.15.2         | 2.3 MB    | ################################################################################################################################################################################################################################### | 100% 
traceback2-1.4.0     | 28 KB     | ################################################################################################################################################################################################################################### | 100% 
chardet-3.0.4        | 180 KB    | ################################################################################################################################################################################################################################### | 100% 
pycparser-2.19       | 87 KB     | ################################################################################################################################################################################################################################### | 100% 
samtools-1.9         | 636 KB    | ################################################################################################################################################################################################################################### | 100% 
pip-19.0.2           | 1.8 MB    | ################################################################################################################################################################################################################################### | 100% 
wheel-0.33.0         | 34 KB     | ################################################################################################################################################################################################################################### | 100% 
cryptography-2.5     | 639 KB    | ################################################################################################################################################################################################################################### | 100% 
h5py-2.8.0           | 1.0 MB    | ################################################################################################################################################################################################################################### | 100% 
setuptools-40.8.0    | 630 KB    | ################################################################################################################################################################################################################################### | 100% 
blasr_libcpp-5.3.1   | 2.0 MB    | ################################################################################################################################################################################################################################### | 100% 
pysocks-1.6.8        | 22 KB     | ################################################################################################################################################################################################################################### | 100% 
numpy-1.16.1         | 4.2 MB    | ################################################################################################################################################################################################################################### | 100% 
backports-1.0        | 4 KB      | ################################################################################################################################################################################################################################### | 100% 
libcurl-7.64.0       | 586 KB    | ################################################################################################################################################################################################################################### | 100% 
cython-0.29.5        | 2.1 MB    | ################################################################################################################################################################################################################################### | 100% 
backports.functools_ | 6 KB      | ################################################################################################################################################################################################################################### | 100% 
idna-2.8             | 133 KB    | ################################################################################################################################################################################################################################### | 100% 
bcftools-1.9         | 813 KB    | ################################################################################################################################################################################################################################### | 100% 
libedit-3.1.20170329 | 172 KB    | ################################################################################################################################################################################################################################### | 100% 
pyopenssl-19.0.0     | 79 KB     | ################################################################################################################################################################################################################################### | 100% 
pytz-2018.9          | 229 KB    | ################################################################################################################################################################################################################################### | 100% 
bzip2-1.0.6          | 415 KB    | ################################################################################################################################################################################################################################### | 100% 
curl-7.64.0          | 143 KB    | ################################################################################################################################################################################################################################### | 100% 
python-dateutil-2.8. | 219 KB    | ################################################################################################################################################################################################################################### | 100% 
krb5-1.16.3          | 1.4 MB    | ################################################################################################################################################################################################################################### | 100% 
linecache2-1.0.0     | 13 KB     | ################################################################################################################################################################################################################################### | 100% 
futures-3.2.0        | 24 KB     | ################################################################################################################################################################################################################################### | 100% 
ruamel_yaml-0.15.71  | 249 KB    | ################################################################################################################################################################################################################################### | 100% 
iso8601-0.1.12       | 9 KB      | ################################################################################################################################################################################################################################### | 100% 
yaml-0.1.7           | 78 KB     | ################################################################################################################################################################################################################################### | 100% 
requests-2.21.0      | 83 KB     | ################################################################################################################################################################################################################################### | 100% 
ipaddress-1.0.22     | 18 KB     | ################################################################################################################################################################################################################################### | 100% 
htslib-1.9           | 1.5 MB    | ################################################################################################################################################################################################################################### | 100% 
conda-env-2.6.0      | 2 KB      | ################################################################################################################################################################################################################################### | 100% 
six-1.12.0           | 21 KB     | ################################################################################################################################################################################################################################### | 100% 
unittest2-1.1.0      | 68 KB     | ################################################################################################################################################################################################################################### | 100% 
hdf5-1.10.3          | 5.3 MB    | ################################################################################################################################################################################################################################### | 100% 
pycosat-0.6.3        | 103 KB    | ################################################################################################################################################################################################################################### | 100% 
python-2.7.15        | 11.9 MB   | ################################################################################################################################################################################################################################### | 100% 
enum34-1.1.6         | 56 KB     | ################################################################################################################################################################################################################################### | 100% 
asn1crypto-0.24.0    | 154 KB    | ################################################################################################################################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: | 
##############################################################################
#                                                                            #
# PacBio(R) tools distributed via Bioconda are: pre-release versions, not    #
# necessarily ISO compliant, intended for Research Use Only and not for use  #
# in diagnostic procedures, intended only for command-line users, and        #
# possibly newer than the currently available SMRT(R) Analysis builds. While #
# efforts have been made to ensure that releases on Bioconda live up to the  #
# quality that PacBio strives for, we make no warranty regarding any         #
# Bioconda release.                                                          #
#                                                                            #
# As PacBio tools distributed via Bioconda are not covered by any service    #
# level agreement or the like, please *do not* contact a PacBio Field        #
# Applications Scientist or PacBio Customer Service for assistance with any  #
# Bioconda release. We instead provide an issue tracker for you to report    #
# issues to us at:                                                           #
#                                                                            #
#   https://github.com/PacificBiosciences/pbbioconda                         #
#                                                                            #
# We make no warranty that any such issue will be addressed,                 #
# to any extent or within any time frame.                                    #
#                                                                            #
# BSD 3-Clause Clear License                                                 #
#                                                                            #
# Please see https://github.com/PacificBiosciences/pbbioconda for            #
# information on License, Copyright and Disclaimer                           #
#                                                                            #
##############################################################################

/ 
##############################################################################
#                                                                            #
# PacBio(R) tools distributed via Bioconda are: pre-release versions, not    #
# necessarily ISO compliant, intended for Research Use Only and not for use  #
# in diagnostic procedures, intended only for command-line users, and        #
# possibly newer than the currently available SMRT(R) Analysis builds. While #
# efforts have been made to ensure that releases on Bioconda live up to the  #
# quality that PacBio strives for, we make no warranty regarding any         #
# Bioconda release.                                                          #
#                                                                            #
# As PacBio tools distributed via Bioconda are not covered by any service    #
# level agreement or the like, please *do not* contact a PacBio Field        #
# Applications Scientist or PacBio Customer Service for assistance with any  #
# Bioconda release. We instead provide an issue tracker for you to report    #
# issues to us at:                                                           #
#                                                                            #
#   https://github.com/PacificBiosciences/pbbioconda                         #
#                                                                            #
# We make no warranty that any such issue will be addressed,                 #
# to any extent or within any time frame.                                    #
#                                                                            #
# BSD 3-Clause Clear License                                                 #
#                                                                            #
# Please see https://github.com/PacificBiosciences/pbbioconda for            #
# information on License, Copyright and Disclaimer                           #
#                                                                            #
##############################################################################

done
```

It does seem to update e.g. numpy. However, when I rerun on the sample data, I am getting the same errors:

```
$ arrow -j 16 -o ref_arrow.fa -r ref.fa subsample.pbalign_test.bam
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
Arrow: unsupported chemistries found: (unknown)
Arrow: unsupported chemistries found: (unknown)
Traceback (most recent call last):
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/pbcommand/cli/core.py", line 138, in _pacbio_main_runner
    return_code = exe_main_func(*args, **kwargs)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 340, in args_runner
    return tr.main()
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 270, in main
    self._configureAlgorithm(options, peekFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 164, in _configureAlgorithm
    self._algorithmConfiguration = self._algorithm.configure(options, alnFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/arrow/arrow.py", line 236, in configure
    die("Arrow: unsupported chemistries found: ({0})".format(", ".join(sorted(unsupp))))
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/utils.py", line 16, in die
    raise DieException(msg)
DieException: Arrow: unsupported chemistries found: (unknown)
[ERROR] Arrow: unsupported chemistries found: (unknown)
Traceback (most recent call last):
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/pbcommand/cli/core.py", line 138, in _pacbio_main_runner
    return_code = exe_main_func(*args, **kwargs)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 340, in args_runner
    return tr.main()
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 270, in main
    self._configureAlgorithm(options, peekFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/main.py", line 164, in _configureAlgorithm
    self._algorithmConfiguration = self._algorithm.configure(options, alnFile)
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/arrow/arrow.py", line 236, in configure
    die("Arrow: unsupported chemistries found: ({0})".format(", ".join(sorted(unsupp))))
  File "/home/ec2-user/miniconda3/lib/python2.7/site-packages/GenomicConsensus/utils.py", line 16, in die
    raise DieException(msg)
DieException: Arrow: unsupported chemistries found: (unknown)
```

Likely I am not understanding a subtlety of conda; it does seem like it is a compatibility issue but I don't know enough about conda to guess what it is.

**Comment by pb-dseifert** [comment]
You have 39k (*!*) contigs, that you want to polish. Without some of them, I can't start debugging Arrow. If you like you can also share them with me by email.

**Comment by GuyAllard** [comment]
I am also getting an error about invalid sequencing chemistry, this time using the bioconda release of LAA version 2.4.2

The error message when running LAA is:
```
>|> 20190221 14:35:25.303 -|- FATAL      -|- operator() -|- 0x7f2fb09e77c0|| -|- caught exception: "unsupported sequencing chemistry combination:
    binding kit:        101-500-400
    sequencing kit:     101-427-800
    basecaller version: 5.0.0
"
>|> 20190221 14:35:25.303 -|- FATAL      -|- operator() -|- 0x7f2fb09e77c0|| -|- caught SIGABRT
Aborted (core dumped)
```

The data which is giving the problem was generated by the newer Sequel sequencer.
Data generated by the older RS2 instrument does not give this error.

If I use LAA version 2.4.2 installed from the latest smrtlink release available from PacBio, the data is processed without error, so it seems specific to the bioconda release.

**Comment by pb-dseifert** [comment]
LAA hasn't been updated yet, this will happen at some point when we have time.

**Comment by pb-dseifert** [comment]
@maximilianpress can you please post the header of the BAM file?

**Comment by maximilianpress** [comment]
Sorry, thought I'd sent a headered bam:
[pbalign_header.txt](https://github.com/PacificBiosciences/pbbioconda/files/2962534/pbalign_header.txt)

**Comment by maximilianpress** [comment]
Incidentally, I just reinstalled following the normal instructions on a fresh Ubuntu environment, and both of my errors went away. I noticed that genomicConsensus version was 2.3.3. So likely the update (possibly new LAA?) fixed it.

Then I went back to my initial Amazon Linux environment and ran `conda update --all`. I found that these Arrow errors went away. However there were new errors, that I solved by getting rid of various system numpys. Under these errors/warnings, Arrow still ran but then I got the same unsupported chemistries error as initially.

Finally I got it using the right numpy and the numpy errors went away. I _think_ it is now working.

Thanks for looking into this.
